### PR TITLE
Worked on fixing issue W0612 (unused-variable) in "common"

### PR DESCRIPTION
**Pull request by dnerever**
Fixed the occurrence of unused variables through multiple files within `/common`

Related to #1755

### CODE REVIEW

File: common/config.py [hunk 1]
```diff
@@ -1603,11 +1603,10 @@ def olderThan(self, time, value, unit):
...
             firstDay = now.date() - datetime.timedelta(days = now.date().day + 1)
-            for i in range(value - 1):
-                if firstDay.month == 1:
-                    firstDay = firstDay.replace(month = 12, year = firstDay.year - 1)
-                else:
-                    firstDay = firstDay.replace(month = firstDay.month - 1)
+            if firstDay.month == 1:
+                firstDay = firstDay.replace(month = 12, year = firstDay.year - 1)
+            else:
+                firstDay = firstDay.replace(month = firstDay.month - 1)
```
Comment (buhtz) [inline common/config.py]: What is the rational behind this? The for loop is gone.

I assume the iteration is needed but the iteration counter not. In this case someone should write

    for _ in range(values-1):
        foobar()

The `_` is an accepted variable name in the Python world for "unnamed" variables.

But this method seems to be very important. We could break a lot if something is wrong here.
I will have a look if there is a good unit test for this method and if not I would write one. Or do you want to write a unit test?

File: common/schedule.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def is_cron_running():
 
     with subprocess.Popen(['ps', '-eo', 'comm'], stdout=subprocess.PIPE) as ps:
         try:
-            grep = subprocess.run(
```
Comment (buhtz) [inline common/schedule.py:178]: Great. This is my own code introduced not long ago. ;)

Absolutely right this modificiation.

File: common/sshtools.py [hunk 3]
```diff
@@ -568,17 +568,16 @@ def benchmarkCipher(self, size=40):
 
             print('%s%s:%s' % (bcolors.BOLD, cipher, bcolors.ENDC))
 
-            for i in range(2):
```
Comment (buhtz) [inline common/sshtools.py]: The whole method is unclear to me what its purpose is.
This for-loop makes no sense to me. But I assume there is one.

If you don't have an idea I would recommend to reset that modification and we remember that in a new issue.

File: common/sshtools.py [hunk 4]
```diff
@@ -835,7 +834,7 @@ def maxArg():
                                         stdout=subprocess.PIPE,
                                         stderr=subprocess.PIPE,
                                         universal_newlines=True)
-                out, err = proc.communicate()
```
Comment (buhtz) [inline common/sshtools.py:843]: You can keep that modification. That is ok.

**Comment by buhtz** [comment]
Hello Keith,
Thank you for that PR. You see it can get complicated. I added some comments. Let me know how you want to proceed.

Because of its complexity and possible impact I schedule that PR until the upcoming release (see milestone).

File: common/config.py (hunk 1 above)
Comment (buhtz) [reply common/config.py]: I introduced #1791 . So just reset this modification and leave it as it is. We will take care of it another time.

**Comment by dnerever** [comment]
Hey Christian,
Can you direct me to where you added your comments. 

I do see how it can get complicated so delaying it makes sense. Any suggestions for ensuring my future PRs avoid introducing complexity? I'm going now to take a look at the other opens issues in the repo.

Thanks again for all the help!

**Comment by buhtz** [comment]
You are not the one making it complicated. The old and smelly Back In Time code does. I just tried to pointing out that even simple modifications can become complicated in old code base like Back In Time has.

About the comments. I added 4 comments directly to the diff-like output. This means a comment is connected directly to a line in the code. I marked them with yellow circles in this screenshot. Just scroll up.

image

**Review by buhtz — APPROVED** [review]

**Comment by buhtz** [comment]
To prevent you from extra work and wasting time, I just want to mention that someone else will work on the same rule but in "qt" folder.

https://github.com/bit-team/backintime/issues/1755#issuecomment-2271453055

EDIT: By the way I realized that you did not activate the W0612 rule in `common/test/test_lint.py` in this PR. Is there are reason? If all W0612 errors in "common" are fixed you can activate it.

**Review by buhtz — CHANGES_REQUESTED** [review]
Please see my last comment.

**Comment by buhtz** [comment]
I see why you didn't enable W0612 by default. There are to many errors.

**Review by buhtz — APPROVED** [review]

**Pull request merged**